### PR TITLE
Edit arena off season

### PR DESCRIPTION
**Pull request by tyrosine1153**
- Hard fork BattleArena from 7 to 8, ArenaHelper
- Delete BattleArena validation(block interval, score difference) in off season
- Edit ticket use
- Update BattleArenaTest

### CODE REVIEW

**Comment by sonohoshi** [comment]
> Hard fork BattleArena from 4 to 5, ArenaHelper
As far as I know, new BattleArena version is 8.

File: .Lib9c.Tests/Action/BattleArenaTest.cs [hunk 1]
```diff
@@ -0,0 +1,1182 @@
...
+    using Nekoyume.Model.Arena;
+    using Nekoyume.Model.EnumType;
+    using Nekoyume.Model.State;
+    using Nekoyume.TableData;
+    using Serilog;
+    using Xunit;
+    using Xunit.Abstractions;
+    using static Lib9c.SerializeKeys;
+
+    public class BattleArenaTest
```
Comment (sonohoshi) [inline .Lib9c.Tests/Action/BattleArenaTest.cs:24]: **OriginActionName**Test was obsoleted. Please writing in [BattleArena8Test.cs](https://github.com/planetarium/lib9c/blob/development/.Lib9c.Tests/Action/BattleArena8Test.cs).

**Review by sonohoshi — APPROVED** [review]

File: .Lib9c.Tests/Action/BattleArena8Test.cs [hunk 2]
```diff
@@ -938,6 +937,83 @@ public void ExecuteDuplicatedException(int slotIndex, int runeId, int slotIndex2
...
+                championshipId = championshipId,
+                round = round,
+                ticket = 8,
+                costumes = new List<Guid>(),
+                equipments = new List<Guid>(),
+                runeInfos = new List<RuneSlotInfo>(),
+            };
+
+            var blockIndex = roundData.StartBlockIndex + 1;
+            Assert.Throws<ValidateDuplicateTicketPurchaseException>(() => action.Execute(new ActionContext
```
Comment (Unengine) [inline .Lib9c.Tests/Action/BattleArena8Test.cs:1008]: I think you should check if current blockIndex is in off-season.

File: Lib9c/Model/Arena/ArenaExceptions.cs [hunk 3]
```diff
@@ -125,6 +125,13 @@ public NotEnoughTicketException(string message) : base(message)
         }
     }
 
+    public class ValidateDuplicateTicketPurchaseException : Exception
```
[the remark this answers is not in the document]
Comment (Unengine) [reply Lib9c/Model/Arena/ArenaExceptions.cs:128]: I recommend `TicketPurchaseLimitExceedException`.

**Review by Unengine — APPROVED** [review]

**Pull request merged**